erpnext/accounts/doctype/promotional_scheme/promotional_scheme.py

```diff
@@ -30,17 +30,24 @@ def validate(self):
 			frappe.throw(_("Price or product discount slabs are required"))
 
 	def on_update(self):
-		data = frappe.get_all('Pricing Rule', fields = ["promotional_scheme_id", "name"],
-			filters = {'promotional_scheme': self.name}) or {}
-
+		data = frappe.get_all(
```

Comment:
`data` that is being passed around is making i hard to comprehend what it contains
```suggestion
		pricing_rules = frappe.get_all(
```